### DISCUSSION TITLE
Request a code to reset password

### DISCUSSION THREAD

**Discussion by gokulraghuthaman** [zitadel/zitadel#7262] [Q&A] [answered]
Hi,
I am using zitadel for the authentication in my project. For integrating the forgot passoword functionality I am using the api for request a code for reset the password
url : {{base_url}}v2beta/users/:userId/password_reset
payload : {
      "sendLink": {
        "notificationType": "NOTIFICATION_TYPE_Unspecified",
        "urlTemplate": "https://example.com/password/changey?userID=248665926554102649"
      },
      "returnCode": {}
    }
 I am getting a bad request error when hitting this api ie
 {
    "code": 3,
    "message": "proto: (line 6:3): error parsing \"returnCode\", oneof zitadel.user.v2beta.PasswordResetRequest.medium is already set"
}
for returnCode in the documentation there is nothing specified, what is the problem in this?

**Comment by hifabienne** [comment, marked as answer]
Hei @gokulraghuthaman 
I think the documentation is not clear about that. You have to choose between send Link or return code.
So if you do want that ZITADEL does send the code, you have to fill the sendLink, and don't send returnCode
If you do want to send the email yourself and get a code as response, you should not send the sendLink

**Comment by gokulraghuthaman** [comment]
Thank you for your replay. This is the correct one, I was confused by the documentation, now its clear.
